src/utils/data-utils/src/data/format.rs

```diff
@@ -33,6 +34,19 @@ pub trait RecordsWriter {
     fn finish(&mut self) -> Result<(), Error> {
         Ok(())
     }
+
+    fn handle_writer_result(&self, writer_result: &Result<(), ArrowError>) -> Result<(), Error> {
+        if let Err(err) = writer_result {
+            match err {
+                ArrowError::IoError(err_str, io_err) => match io_err.kind() {
+                    ErrorKind::BrokenPipe => (),
+                    _ => panic!("Cannot write output, io error occurred : {err_str}"),
+                },
+                err => panic!("Cannot write output, arrow writer error occurred: {err}"),
+            };
+        }
+        Ok(())
```

Comment:
I don't think it's helpful to panic.
Perhaps, extend the error type for `handle_writer_result` to become an error-enum, and convert the result into managed errror returned to commands.